### DISCUSSION TITLE
Proposal: syntatic sugar keyword num or number

### DISCUSSION THREAD

**Discussion by avertiy** [dotnet/csharplang#5577] [Language Ideas]
Syntatic sugar keyword **num**  (from number)

`
num Multiply(num a, num b) => a*b; 
`

currently developer has to add manually overloaded versions for each primitive type 

the suggestion is to delegate this job to compilator so depending on usages of the Multiply method compiler will generate all the required overloaded versions with short, int, long, double etc. types.

**Comment by CyrusNajmabadi** [comment]
This can be done with generics and `INumber<T>` :-)
